### PR TITLE
Make it compile against Jellyfin

### PR DESCRIPTION
**Pull request by hawken93**
Needs Jellyfin.Naming (Emby.Naming) to compile but it doesn't exist currently afaik

### CODE REVIEW

File: Emby.AutoOrganize/Emby.AutoOrganize.csproj [hunk 1]
```diff
@@ -35,8 +35,8 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="MediaBrowser.Naming" Version="1.1.7-beta" />
-    <PackageReference Include="mediabrowser.server.core" Version="3.6.0.43-beta" />
+    <PackageReference Include="Jellyfin.Common" Version="10.0.1" />
```
Comment (Bond-009) [inline Emby.AutoOrganize/Emby.AutoOrganize.csproj]: Common is included in Controller

Comment (hawken93) [reply Emby.AutoOrganize/Emby.AutoOrganize.csproj]: "error CS0234: The type or namespace name 'Controller' does not exist in the namespace 'MediaBrowser' (are you missing an assembly reference?)"

this disappeared after I added that line...

Comment (hawken93) [reply Emby.AutoOrganize/Emby.AutoOrganize.csproj]: Never mind! :)

**Comment by hawken93** [comment]
~~https://github.com/jellyfin/jellyfin/pull/587~~ - PR to get Jellyfin.Naming out there

**Comment by hawken93** [comment]
I didn't rewrite the code fully yet, I expect compile errors when the packages are up but I will continue then :)

**Comment by joshuaboniface** [comment]
@hawken93 Still a WIP?

**Comment by hawken93** [comment]
@joshuaboniface I haven't had time to look at this and hence it's probably still not compiling. If you want we can clean this up and merge so it's at least a bit closer to compiling, if someone has time to take this further then I would suggest to do that.

**Comment by hawken93** [comment]
@joshuaboniface It's way closer to compiling now, but I need some help with the remaining ones.. I cleaned up the logging part and updated to compile against 10.1.0. @Bond-009 maybe I can ask for your help here

**Comment by hawken93** [comment]
~Errors:~

Warnings:
```
Data/SqliteFileOrganizationRepository.cs(95,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Data/SqliteFileOrganizationRepository.cs(118,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Data/SqliteFileOrganizationRepository.cs(134,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/MovieFileOrganizer.cs(273,28): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/EpisodeFileOrganizer.cs(459,28): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Data/SqliteFileOrganizationRepository.cs(367,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
```

**Comment by hawken93** [comment]
This is going to be a big PR because of reformatting and stuff but I try my best to keep each commit very targeted, so I recommend to review commit by commit

**Comment by hawken93** [comment]
```
commit 391b48614d598b5fbfad68b610520a5ae7a57b70
Author: Bond_009 <bond.009@outlook.com>
Date:   Fri Dec 28 15:21:02 2018 +0100

    Remove FireEventIfNotNull
    
    It's a pretty useless "helper" class
```
:/ ok so how do I do this..

**Comment by hawken93** [comment]
The remaining errors seem to be caused by GPL cleanup. I'd fix the warnings but I'm not familiar enough with C#s async logic

**Comment by hawken93** [comment]
Compiling! Thanks @cvium!

**Comment by hawken93** [comment]
Decided to leave the warnings for the c# devs with skillz to sort out. It's not critical

**Comment by hawken93** [comment]
@joshuaboniface ping, I'm ready

**Review by cvium — APPROVED** [review]
A lot of the logger statements should be cleaned up, but I think it's good enough for now -- don't want you to get all depressed working on the plugins lol. We should have an issue to track the async warnings and general cleanup though.

**Comment by hawken93** [comment]
I want to get rid of the general exception catchers, but I know it will break stuff

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**